tests/integration/cocoon/test_reference_portfolios.py

```diff
@@ -0,0 +1,132 @@
+        ]
+    )
+    def test_load_from_data_frame_a_portfolios_failure(
+        self,
+        _,
+        file_name,
+        mapping_required,
+        mapping_optional,
+        property_columns,
+        expected_outcome,
+    ) -> None:
+        """
+        Test that a reference portfolio can be loaded successfully
+
+        :param str file_name: The name of the test data file
+        :param dict{str, str} mapping_required: The dictionary mapping the dataframe fields to LUSID's required base transaction/holding schema
+        :param dict{str, str} mapping_optional: The dictionary mapping the dataframe fields to LUSID's optional base transaction/holding schema
+        :param list[str] property_columns: The columns to create properties for
+        :param any expected_outcome: The expected outcome
+
+        :return: None
+        """
+
+        unique_id = create_scope_id()
+        data_frame = pd.read_csv(Path(__file__).parent.joinpath(file_name))
+        data_frame["FundCode"] = data_frame["FundCode"] + "-" + unique_id
+
+        responses = cocoon.cocoon.load_from_data_frame(
+            api_factory=self.api_factory,
+            scope=self.scope,
+            data_frame=data_frame,
+            mapping_required=mapping_required,
+            mapping_optional=mapping_optional,
+            file_type="reference_portfolios",
+            identifier_mapping={},
+            property_columns=property_columns,
+            properties_scope=self.scope,
+            sub_holding_keys=[],
+        )
+
+        # Check that the count of portfolios uploaded equals count of portfolios in DataFrame
+
+        self.assertEqual(
+            first=len(responses["reference_portfolios"]["success"]),
+            second=len(data_frame),
+        )
+
+        # Check that the portfolio IDs of porfolios uploaded matches the IDs of portfolios in DataFrame
+
+        response_codes = [
+            response.id.code
+            if isinstance(response, lusid.models.Portfolio)
+            else response.origin_portfolio_id.code
+            for response in responses["reference_portfolios"]["success"]
+        ]
+
+        self.assertEqual(
+            first=response_codes,
+            second=list(data_frame[mapping_required["code"]].values),
+        )
+
+        # Check that properties get added to portfolio
+
+        for portfolio in responses["reference_portfolios"]["success"]:
+
+            get_portfolio = self.portfolios_api.get_portfolio(
+                scope=portfolio.id.scope,
+                code=portfolio.id.code,
+                property_keys=[
+                    f"Portfolio/{self.scope}/strategy",
+                    f"Portfolio/{self.scope}/custodian",
+                ],
+            )
+
+            property_keys_from_params = [
+                f"Portfolio/{self.scope}/{code}"
+                for code in property_columns
+                if len(property_columns) > 0
+            ]
+
+            self.assertListEqual(
```

Comment:
i don't think order is guaranteed, this may need to be converted to a `set` and `assertSetEquals`

Reply:
Sure.
Or we could use assertCountEqual.
I have just read this will compare two lists regardless of order which is what we want. 
https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertCountEqual

Reply:
that would work too, I thought assertCountEqual would, well check the counts were the same but if checks the values too then perfect!

Reply:
Thanks. It appears to check **count** and **contents** which is what we want (agreed name is slightly misleading).